modules/govuk_elasticsearch/manifests/snapshot.pp

```diff
@@ -0,0 +1,66 @@
+# define govuk_elasticsearch::snapshot
+#
+# [*s3_bucket]
+#   The S3 bucket where the snapshots will be stored
+#
+# [*s3_region]
+#   The AWS region for the S3 bucket. Defaults to eu-west-1.
+#
+# [*s3_path]
+#   The folder path within the S3 bucket where the snapshots
+#   will be stored
+#
+# [*index]
+#   The ES index to backup
+#
+# [*http_port]
+#   The port that ES runs on
+#
+# [*hour]
+#   The hour to start the snapshot (crontab)
+#
+# [*minute]
+#   The minute to run the snapshot (crontab)
+#
+# [*title*]
+#   The $title or $name of the hiera hash should be the index name
+#
+# Hiera example:
+# --------------
+# govuk_elasticsearch::snapshot::indices:
+#   kibana-int:
+#     s3_bucket: 'jonauman-test2'
+#     s3_path: 'elasticsearch'
+
+define govuk_elasticsearch::snapshot (
+  $s3_bucket = undef,
+  $s3_region = 'eu-west',
+  $s3_path = undef,
+  $index = $title,
+  $http_port = 9200,
+  $hour = 3,
+  $minute = 20,
+  ){
+
+  file { "/usr/local/bin/elasticsearch-create-repo-${index}":
+    ensure  => present,
+    content => template('govuk_elasticsearch/create_repo.sh.erb'),
+  }
+
+  file { "/usr/local/bin/elasticsearch-backup-${index}":
+    ensure  => present,
+    content => template('govuk_elasticsearch/backup_index.sh.erb'),
+  }
+
+  exec { "/usr/local/bin/elasticsearch-create-repo-${index}":
```

Comment:
This script has to be run at least once to setup the S3 repository, though it doesn't hurt to run it with each puppet run. So cron didn't seem to be like a good way to get it executed.

I suppose I could add it to the backup script. Thoughts?